### PR TITLE
Plugin configuration UI, full page error

### PR DESCRIPTION
**Pull request by brojd**
# What this PR does

## Which issue(s) this PR closes

Closes [issue link here]

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx [hunk 1]
```diff
@@ -1,54 +1,241 @@
...
-        )}
-      />
-      <HorizontalGroup>
-        <Button type="submit" disabled={!formState.isValid}>
-          Test & Save connection
-        </Button>
-        {config.featureToggles.externalServiceAccounts && <Button variant="secondary">Recreate service account</Button>}
-      </HorizontalGroup>
-    </form>
+    <>
```
Comment (teodosii) [inline grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx:39]: shouldn't this be wrapped in like a `VerticalGroup` at least?

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx [hunk 2]
```diff
@@ -1,54 +1,241 @@
...
+
+    return (
+      <>
+        <Text type="secondary" className={styles.secondaryTitle}>
+          This page will help you to connect OnCall backend and OnCall Grafana plugin.
+        </Text>
+        <CollapsibleTreeView
+          className={styles.treeView}
+          configElements={[
+            {
```
Comment (teodosii) [inline grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx:96]: maybe worth splitting these items somehow? seems like a lot of nesting/conditionals

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx [hunk 3]
```diff
@@ -1,54 +1,241 @@
...
+
+const PluginConfigAlert = observer(() => {
+  const {
+    pluginStore: { connectionStatus, isPluginConnected },
+  } = rootStore;
+
+  if (!connectionStatus) {
+    return null;
+  }
+  return isPluginConnected ? (
```
Comment (teodosii) [inline grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx:199]: IMO it's cleaner if you don't nest here and instead use an if on the previous line.

File: grafana-plugin/src/models/plugin/plugin.ts [hunk 4]
```diff
@@ -1,48 +1,64 @@
...
+import { PluginConnection, PostStatusResponse } from 'network/oncall-api/api.types';
 import { RootBaseStore } from 'state/rootBaseStore/RootBaseStore';
 import { AutoLoadingState } from 'utils/decorators';
-import { getIsRunningOpenSourceVersion } from 'utils/utils';
+
+import { PluginHelper } from './plugin.helper';
+
+/* 
+High-level OnCall initialization process:
+On OSS:
```
Comment (teodosii) [inline grafana-plugin/src/models/plugin/plugin.ts:15]: +1 for this

**Review by teodosii — APPROVED** [review]

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx (hunk 1 above)
Comment (brojd) [reply grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx:39]: Doesn't harm, let me do that. It gives a bit more spaces between particular parts

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx (hunk 3 above)
Comment (brojd) [reply grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx:201]: Not sure if we can get better here, let's discuss on Slack

File: grafana-plugin/src/models/plugin/plugin.ts [hunk 5]
```diff
@@ -1,48 +1,64 @@
...
+    });
+    runInAction(() => {
+      this.connectionStatus = pluginConnection;
+      this.isPluginConnected = Object.keys(pluginConnection).every(
+        (key) => pluginConnection[key as keyof PluginConnection]?.ok
+      );
+    });
+  }
+
+  @AutoLoadingState(ActionKey.UPDATE_SETTINGS_AND_REINITIALIZE_PLUGIN)
```
[the remark this answers is not in the document]
Comment (brojd) [reply grafana-plugin/src/models/plugin/plugin.ts]: done

File: grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx (hunk 1 above)
Comment (brojd) [reply grafana-plugin/src/containers/PluginConfigPage/PluginConfigPage.tsx]: done

**Pull request merged**